python/ray/data/datasource/parquet_datasink.py

```diff
@@ -1,20 +1,25 @@
-import logging
-from typing import TYPE_CHECKING, Any, Callable, Dict, Optional
+import posixpath
+from typing import TYPE_CHECKING, Any, Callable, Dict, Iterable, Optional
 
-from ray.data.block import BlockAccessor
+from ray.data._internal.dataset_logger import DatasetLogger
+from ray.data._internal.execution.interfaces import TaskContext
+from ray.data._internal.util import call_with_retry
+from ray.data.block import Block, BlockAccessor
+from ray.data.context import DataContext
 from ray.data.datasource.block_path_provider import BlockWritePathProvider
-from ray.data.datasource.file_based_datasource import _resolve_kwargs
-from ray.data.datasource.file_datasink import BlockBasedFileDatasink
+from ray.data.datasource.file_datasink import _FileDatasink
 from ray.data.datasource.filename_provider import FilenameProvider
 
 if TYPE_CHECKING:
     import pyarrow
 
+WRITE_FILE_MAX_ATTEMPTS = 10
+WRITE_FILE_RETRY_MAX_BACKOFF_SECONDS = 32
 
-logger = logging.getLogger(__name__)
+logger = DatasetLogger(__name__)
 
 
-class _ParquetDatasink(BlockBasedFileDatasink):
+class _ParquetDatasink(_FileDatasink):
```

Comment:
@raulchen decided to subclass `_FileDatasink` rather than change `write_block_to_file` to minimize API churn (it's not a public API, but it's still a documented developer API).